Resources/public/js/views/ez-testonearticlesview.js

```diff
@@ -0,0 +1,195 @@
+    Y.namespace('eZ');
+
+    var ARTICLE_LINK_SELECTOR = '[data-location-id="{locationId}"]',
+        ARTICLES_CLASS_PREFIX = 'ez-view-testonearticlesview',
+        LOADER = Y.Node.create('<span class="ez-inline-loader"></span>');
+
+    /**
+     * The article list view
+     *
+     * @namespace eZ
+     * @class TestOneArticlesView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.TestOneArticlesView = Y.Base.create('testOneArticlesView', Y.eZ.TemplateBasedView, [], {
+        initializer: function () {
+            this.after('articlesChange', this.render);
+            this.on('rootLocationChange', this._changeRootLocation);
+            this.on('*:articleLoading', this._articleLoading);
+            this.on('*:articleLoaded', this._articleLoaded);
+        },
+
+        render: function () {
+            var history = this.get('history') || [],
+                currentLocation = this.get('currentLocation');
+
+            this.get('container').setHTML(this.template({
+                articles: this.get('articles'),
+                previous: history.length > 0 ? history[history.length - 1].article : null,
+                currentLocation: currentLocation ? currentLocation.name : null
+            }));
+
+            this.get('container').all('[data-location-id]').on('click', this._selectArticle, this);
+
+            return this;
+        },
+
+        /**
+         * Event handler for starting article list loading
+         *
+         * @method _articleLoading
+         * @param {Object} e event facade
+         * @protected
+         */
+        _articleLoading: function (e) {
+            var container = this.get('container');
+            container.addClass(ARTICLES_CLASS_PREFIX + '--loading');
+            var articleLink = container.one(Y.Lang.sub(ARTICLE_LINK_SELECTOR, {
+                locationId: e.locationId
+            }));
+
+            if (articleLink) {
+                articleLink.append(LOADER);
+            } else {
+                container.append(LOADER);
+            }
+        },
+
+        /**
+         * Event handler for article loaded
+         *
+         * @method _articleLoaded
+         * @param {Object} e event facade
+         * @protected
+         */
+        _articleLoaded: function (e) {
+            this.set('currentLocation', e.currentLocation);
+            this.get('container').removeClass(ARTICLES_CLASS_PREFIX + '--loading');
+            LOADER.detach();
+            this.set('articles', e.articles);
+        },
+
+        /**
+         * Event handler for selecting article link
+         *
+         * @method _selectArticle
+         * @param {Object} e event facade
+         * @protected
+         */
+        _selectArticle: function (e) {
+            e.preventDefault();
+            var selectedArticle = this.get('selectedArticle'),
+                articles = this.get('articles'),
+                locationId = e.target.getAttribute('data-location-id'),
```

Comment:
Nazwy atrybutów, nazwy klas, selektory CSS przenieś do _stałych_ zmiennych. Łatwiej jest wtedy zarządzać kodem. To się odnosi całego kodu JS, który napisałeś. Nie chciałbym się powtarzać, bo w wielu miejscach jest ten problem.